### PR TITLE
Error message and comment improvements 

### PR DESCRIPTION
**Pull request by kevlar700**
Inform the user that watches must occur within a watching_widget within error messages.

### CODE REVIEW

**Comment by escamoteur** [comment]
Thanks a lot, I will review the PR over the weekend

Thomas
Am 29. Juni 2023, 15:49 +0200 schrieb Kevin Chadwick ***@***.***>:

File: lib/src/watch_it.dart [hunk 1]
```diff
@@ -80,7 +80,7 @@ final di = GetIt.I;
 /// this one by manually providing the Listenable that should be observed.
 T watch<T extends Listenable>(T target) {
   assert(_activeWatchItState != null,
-      'watch can only be called inside a build function');
+      'watch can only be called inside a build function within a WatchingWidget');
```
Comment (escamoteur) [inline lib/src/watch_it.dart]: or a widget using the WhatchItMixin

File: lib/src/watch_it.dart [hunk 2]
```diff
@@ -94,7 +94,7 @@ T watch<T extends Listenable>(T target) {
 /// default one. 99% of the time you won't need this.
 T watchIt<T extends Listenable>({String? instanceName, GetIt? getIt}) {
   assert(_activeWatchItState != null,
-      'watch can only be called inside a build function');
+      'watch can only be called inside a build function within a WatchingWidget');
```
Comment (escamoteur) [inline lib/src/watch_it.dart]: sames as above also name the WatchItMixin

File: lib/src/watch_it.dart [hunk 3]
```diff
@@ -118,7 +118,7 @@ T watchIt<T extends Listenable>({String? instanceName, GetIt? getIt}) {
 R watchValue<T extends Object, R>(ValueListenable<R> Function(T) selectProperty,
     {String? instanceName, GetIt? getIt}) {
   assert(_activeWatchItState != null,
-      'watch can only be called inside a build function');
+      'watch can only be called inside a build function within a WatchingWidget');
```
Comment (escamoteur) [inline lib/src/watch_it.dart]: same as above

File: lib/src/watch_it.dart [hunk 4]
```diff
@@ -147,7 +147,7 @@ R watchValue<T extends Object, R>(ValueListenable<R> Function(T) selectProperty,
 R watchPropertyValue<T extends Listenable, R>(R Function(T) selectProperty,
     {T? target, String? instanceName, GetIt? getIt}) {
   assert(_activeWatchItState != null,
-      'watchIt can only be called inside a build function');
+      'watchIt can only be called inside a build function within a WatchingWidget');
```
Comment (escamoteur) [inline lib/src/watch_it.dart]: same as above

**Review by escamoteur — CHANGES_REQUESTED** [review]
Some small things, but many, many thanks for the improvements.
I merge when you fix the requested changes.

**Comment by escamoteur** [comment]
I know its only a nit but because how PRs work, it would be nice if you could add the small changes

**Comment by kevlar700** [comment]
I don't believe I got an email until just now. I shall try to sort this later today.

**Pull request merged**